src/argilla/server/schemas/v1/datasets.py

```diff
@@ -714,8 +726,44 @@ def check_required(cls, values: dict) -> dict:
         return values
 
 
+FilterField = Annotated[str, PydanticField(..., min_length=FILTER_FIELD_MIN_LENGTH, max_length=FILTER_FIELD_MAX_LENGTH)]
+
+
+class TermsFilter(BaseModel):
+    type: Literal["terms"]
+    field: FilterField
+    values: List[str] = PydanticField(
+        ..., min_items=TERMS_FILTER_VALUES_MIN_ITEMS, max_items=TERMS_FILTER_VALUES_MAX_ITEMS
+    )
+
+
+class RangeFilter(BaseModel):
+    type: Literal["range"]
+    field: FilterField
+    gte: float
+    lte: float
```

Comment:
I see this now. We should be able to let users providing only one value (gte or lte) for open ranges

Reply:
Just added the validation for that.